src/assets.rs

```diff
@@ -183,42 +178,42 @@ impl<'a> AssetEditor<'a> {
     ///
     /// Returns the stored asset.
     pub fn save<L: Ledger>(&mut self) -> Result<Asset, KeystoreError<L>> {
-        self.update();
+        if let Ok(store_asset) = self.store.load(&self.asset.definition.code) {
+            self.update(store_asset);
+        }
         if !self.asset.temporary {
             self.store.store(&self.asset.definition.code, &self.asset)?;
         }
         Ok(self.asset.clone())
     }
 
-    /// If there exists an asset in the store with the same code as the editor asset, updates the
-    /// editor asset by merging in the store asset.
-    /// * Keeps the asset name, description or icon of the editor asset if
-    ///   * the editor asset is verified and the store asset is not, and
+    /// Updates the asset by merging in the given asset with the same definition.
+    /// * Keeps the asset name, description or icon if
+    ///   * the existing asset is verified and the other asset is not, and
     ///   * the corresponding attribute exists.
-    /// * Keeps the mint information of the editor asset, if present.
+    /// * Keeps the mint information, if present.
     /// * Sets as verified if either asset if verified.
     /// * Sets as temporary if both assets are temporary.
-    pub fn update(&mut self) {
-        if let Ok(store_asset) = self.store.load(&self.asset.definition.code) {
-            let mut asset = store_asset;
-            if self.asset.verified || !asset.verified {
-                if let Some(name) = self.asset.name.clone() {
-                    asset.name = Some(name);
-                }
-                if let Some(description) = self.asset.description.clone() {
-                    asset.description = Some(description);
-                }
-                if let Some(icon) = self.asset.icon.clone() {
-                    asset.icon = Some(icon);
-                }
+    pub fn update(&mut self, other: Asset) {
```

Comment:
I think it makes more sense for this to be right-biased. So if both `self` and `other` have a certain field, `self` gets updated with the field from `other`, as the name suggests. I believe currently it is the other way around.

Reply:
The current `fn update(&mut self, other: Asset)` takes `self.asset` as the *new asset* and the store asset (given by the `other` param) as the *existing asset*. If they both have a certain field, we take the field of `self.asset`.

`pub fn update(&mut self, info: AssetInfo)` in `asset_library.rs`, however, takes `self.assets` as the *existing asset* and the given `info` as the *new asset*. If they both have a certain field, we take the field of `info`.

So the first one is left-biased and the other is right-biased, but they actually have the same logic, i.e., overriding the existing asset (the store asset in the first case, and `self.assets` in the second case) if both assets have a certain field.

There are two ways to make this clearer: 
1. Rename the function parameter to `fn update(&mut self, existing_asset: Asset)` to make it clear that the given asset is the one already in store. 
2. Keep the name, but take `self.store` as the *existing asset* and `other` as the new asset (so we don't use `self.asset`), and change the function to be right-biased.

From your comment I think you were suggesting the second approach? But would like to confirm if this indeed is what you meant.

Reply:
I think I was suggesting 2? To me, the thing that makes most sense is to take `self.asset` as the existing asset, and update it with new information from `other`. Since `a.update(b)` suggests that `a` is going to be modified by taking information from `b`. Not entirely sure what you mean that "`self.asset` is not used"

Reply:
Oh, I see where the confusion is. I thought when `self.asset` and the store asset both had a certain field, we would take the field of `self.asset` since the store asset was the existing asset. But you were saying the store asset was actually the `other` asset? In this case, then yes, the function should be right-biased.